fix(dashboards): Fixed add widget from discover to new dashboard not creating widget

edwardgou-sentry opened the pull request:
Clicking `Add Widget` on a saved query from the discover landing page and selecting `Create New Dashboard` does not add the widget to the create dashboard view. This pr fixes that bug.

Updated the create view to use `useState` and added `newWidget` handling to dashboard `componentDidUpdate` as well.

shruthilayaj reviewed (approved):
Do you know what caused the regression? Should we add tests too?

edwardgou-sentry commented:
> Do you know what caused the regression? Should we add tests too?

Hey Shruthi, added tests! Not sure exactly what caused the regression but the code to create the new widget from the query parameters was a little fragile. I've updated the `componentDidUpdate` and `componentOnMount` to be more robust.
